crates/rpc/src/context.rs

```diff
@@ -122,7 +122,7 @@ impl RpcContext {
 
     pub fn with_eth_gas_price(self, gas_price: gas_price::Cached) -> Self {
         Self {
-            eth_gas_price: Some(gas_price),
+            eth_gas_price: gas_price,
             ..self
         }
     }
```

Comment:
If we remove the `Option` this method no longer makes sense -- there's no way to construct an RpcContext without a gas price source and calling `RpcContext::new()` is equivalent to what `pathfinder/main.rs` does now with `with_eth_gas_price()`.